### DISCUSSION TITLE
Two node clustering not stopping concurrent jobs running (using fluent, worked without).

### DISCUSSION THREAD

**Discussion by HerbertScroggins** [quartznet/quartznet#2072] [Q&A]
Hi.  I have a two node clustering setup (test harness) which worked beautifully across two nodes (so db is good).

I switched from the previous, older syntax for setting up and configuring to the fluent approach so I could use DI in preparation for moving the code into the main project, but have found that the same jobs are running concurrently on both nodes at the same time, which they weren't previously.  They're just simple tasks  with a timer sleep, to prove the mechanism.

When I start and stop the nodes I see this in the logs...

```
info: Quartz.Impl.StdSchedulerFactory[0]
      Quartz Scheduler 3.6.3.0 - 'node1' with instanceId 'NON_CLUSTERED' initialized info: Quartz.Impl.StdSchedulerFactory[0]
      Using thread pool 'Quartz.Simpl.DefaultThreadPool', size: 3 info: Quartz.Impl.StdSchedulerFactory[0]
      Using job store 'Quartz.Impl.AdoJobStore.JobStoreTX', supports persistence: True, clustered: True info: Microsoft.Hosting.Lifetime[0]
info: Microsoft.Hosting.Lifetime[0]
      Application is shutting down... info: Quartz.Core.QuartzScheduler[0]

...data...

info: Microsoft.Hosting.Lifetime[0]
      Application is shutting down...
info: Quartz.Core.QuartzScheduler[0]
      Scheduler node1_$_NON_CLUSTERED shutting down.
info: Quartz.Core.QuartzScheduler[0]
      Scheduler node1_$_NON_CLUSTERED paused.
info: Quartz.Core.QuartzScheduler[0]
      Scheduler node1_$_NON_CLUSTERED Shutdown complete.

```
My setup code is...

```
services.AddQuartz(q =>
{
    // Add dependency injection.
    q.UseMicrosoftDependencyInjectionJobFactory(options =>
    {
    });

    q.SchedulerName = node;

    q.UsePersistentStore(s =>
    {
        s.PerformSchemaValidation = true;
        s.UseProperties = true; 

        q.UseDefaultThreadPool(tp =>
        {
            tp.MaxConcurrency = 3;                                
        });

        s.UseSqlServer(sqlServer =>
        {
            sqlServer.ConnectionString = "Server=localhost;Database=quartz;Trusted_Connection=True;TrustServerCertificate=true";   
            sqlServer.TablePrefix = "QRTZ_";
        });

        s.UseJsonSerializer();
        s.UseClustering(c =>
        {
            c.CheckinMisfireThreshold = TimeSpan.FromSeconds(30);      
            c.CheckinInterval = TimeSpan.FromSeconds(10);              
        });
    });
   services.AddQuartzHostedService(options =>
   {
       options.WaitForJobsToComplete = true;
   });
})
.Build()
.RunAsync();

```
All jobs are setup with the .DisallowConcurrentExecution() and the [DisallowConcurrentExecution] attribute.

I originally tested this by starting the project in VS and running the exe at the same time (therefore two nodes), and setting the node name differently in each - worked great.  I cannot see, with fluent, how to set the InstanceId to AUTO which appeared necessary in the old syntax to make it work  Of course, I could be completely on the wrong tack here.

Can anyone see what I've missed in configuring it please?

**Comment by HerbertScroggins** [comment]
Interesting, despite both nodes having different values for the `node` variable, if I remove the line `q.SchedulerName = node;` it seems to behave.  Not sure what is going on here but might have made progress.

Yep, that seems to have fixed it.  I still get the logs showing NON_CLUSTERED in the name and I'd love to know why I'm seeing this behaviour but I'm a happy camper once more.

I LOVE this product.

**Comment by lahma** [comment]
https://stackoverflow.com/questions/76667270/getting-concurrent-jobs-running-in-quartz-net-with-clustering-set-to-disallowed

**Reply by HerbertScroggins** [reply]
Yeah sorry about that, posted it to SO before I thought of coming here.  Would appreciate understanding the behaviour though, if you have the time?

**Reply by lahma** [reply]
Maybe you can create a public repository that shows the problem.

**Reply by HerbertScroggins** [reply]
Will do.  Cheers.
